src/expr/src/scalar/func/impls/int32.rs

```diff
@@ -148,3 +148,22 @@ sqlfunc!(
         RegType(a)
     }
 );
+
+sqlfunc!(
+    fn chr(a: i32) -> Result<String, EvalError> {
+        if a < 0 {
+            // This error matches the behavior of Postgres 13/14 (and potentially earlier versions)
+            // Postgres 15 will have a different error message for negative values
+            Err(EvalError::CharacterTooLargeForEncoding(a))
+        } else if a == 0 {
+            Err(EvalError::NullCharacterNotPermitted)
+        } else if 0xd800 <= a && a < 0xe000 {
+            // Postgres returns a different error message for inputs in this range
+            Err(EvalError::CharacterNotValidForEncoding(a))
+        } else {
+            char::from_u32(a as u32)
```

Comment:
nit: We've been migrating as many `as` conversions to `try_into().expect("foo")` as possible, so would prefer that here

Reply:
Just pushed a new version following Nikhil's suggestion, thanks!